toast-tk-web-agent/src/main/java/io/toast/tk/agent/ui/MainApp.java

```diff
@@ -11,246 +11,270 @@
+				InputStream onlineImageAsStream = RestRecorderService.class.getClassLoader().getResourceAsStream("ToastLogo_on.png");   
+				this.onlineImage = ImageIO.read(onlineImageAsStream);
 
-			    PopupMenu popup = new PopupMenu();
+			    PopupMenu popup = initMenuItem();
 			    
-			    MenuItem killItem = new MenuItem("Kill agent");
-			    MenuItem connectItem = new MenuItem("Connect to WebApp");
-			    MenuItem startRecordingItem = new MenuItem("Start Recording");
-			    MenuItem stopRecordingItem = new MenuItem("Stop Recording");
-			    MenuItem settingsItem = new MenuItem("Settings");
-			    
-			    killItem.addActionListener(getKillListener());
-			    connectItem.addActionListener(getConnectListener());
-			    stopRecordingItem.addActionListener(getStopListener());
-			    startRecordingItem.addActionListener(getStartListener());
-			    settingsItem.addActionListener(getSettingsListener());
-			    
-			    popup.add(connectItem);
-			    popup.addSeparator();
-			    popup.add(startRecordingItem);
-			    popup.add(stopRecordingItem);
-			    popup.addSeparator();
-			    popup.add(killItem);
-			    popup.addSeparator();
-			    popup.add(settingsItem); 
-			    
-			    this.trayIcon = new TrayIcon(offline_image, "Toast TK - Web Agent", popup);
-			} catch (IOException e1) {
-				LOG.info(e1);
+			    this.trayIcon = new TrayIcon(offlineImage, "Toast TK - Web Agent", popup);
+				 trayIcon.setImageAutoSize(true);
+				 tray.add(trayIcon);
+			} catch (IOException|AWTException e1) {
+				LOG.error(e1);
 			}
-		    trayIcon.setImageAutoSize(true);
-		    try {
-		        tray.add(trayIcon);
-		    } catch (AWTException e) {
-		    	LOG.info(e);
-		    }
 		}
 	}
 	
-	ActionListener getConnectListener(){
+	private PopupMenu initMenuItem() {
+		PopupMenu popup = new PopupMenu();
+	    
+	    MenuItem quitItem = new MenuItem("Quit");
+	    MenuItem connectItem = new MenuItem("Connect");
+	    MenuItem executeItem = new MenuItem("Execute Scripts");
+	    MenuItem startRecordingItem = new MenuItem("Start Recording");
+	    MenuItem stopRecordingItem = new MenuItem("Stop Recording");
+	    MenuItem settingsItem = new MenuItem("Settings");
+	    
+	    quitItem.addActionListener(getKillListener());
+	    connectItem.addActionListener(getConnectListener());
+	    executeItem.addActionListener(getExecuteListener());
+	    stopRecordingItem.addActionListener(getStopListener());
+	    startRecordingItem.addActionListener(getStartListener());
+	    settingsItem.addActionListener(getSettingsListener());
+	    
+	    popup.add(connectItem);
+	    popup.addSeparator();
+	    popup.add(startRecordingItem);
+	    popup.add(stopRecordingItem);
+	    popup.addSeparator();
+	    popup.add(executeItem);
+	    popup.addSeparator();
+	    popup.add(settingsItem);
+	    popup.addSeparator();
+	    popup.add(quitItem); 
+	    
+	    return popup;
+	}
+	
+	private ActionListener getConnectListener(){
 	    ActionListener listener = new ActionListener() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Make this anonymous inner class a lambda [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1604)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "listener". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1488)